modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/numberShorten.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/*
+ * If a number is bigger than 10000 it will transform into thousands (K).
+ * If a number is bigger than 1000000 it will transform into millions (M).
+ * If a number is bigger than 1000000000 it will transform into billions (B).
+ *
+ * 4 => 4
+ * 4000 => '4K'
+ * 4000000 => '4M'
+ * 4000000000 => '4B'
+ */
+export function numberShorten(value) {
+	if (value >= 1000000000) {
+		return value / 1000000000 + 'B';
```

Comment:
So all these are going to return floats. It's not really going to "shorten" the number at all.

eg. `4298123471 / 1000000000` is `4.298123471`, and if you add `B` to that, you've actually made it longer, not shorter.

Also, you're concatenating a number and a string, and TypeScript will yell at you for that, so it would be better to avoid the implicit coercion.

Did you consider something like?

```js
return value / 1000000000).toFixed(0) + 'B';
```

which is a nicer way to write:

```js

Math.floor(4298123471 / 1000000000).toString() + 'B';
```

tl;dr: I think you should make this function actually shorten the number, or if that is not the intended behavior, rename the function to describe what it does.

Reply:
I'm not controlling the `value` that the function is receiving. It's automatically calculated by Recharts to display the ticks in the chart. I think I'm going with this to prevent longer numbers instead of shortening:

```
return value / 1000000000).toFixed(0) + 'B';
```

Reply:
Sounds good.